src/appleseed/renderer/modeling/bssrdf/directionaldipolebssrdf.h

```diff
@@ -60,14 +60,41 @@ namespace renderer
 // Directional dipole BSSRDF input values.
 //
 
-// The layout of DirectionalDipoleBSSRDFInputValues matches the layout of BSSRDFInputValues.
-// In evaluate_inputs, these values could be converted to BSSRDFInputValues in place.
 APPLESEED_DECLARE_INPUT_VALUES(DirectionalDipoleBSSRDFInputValues)
 {
-    Spectrum    m_diffuse;
-    Spectrum    m_mfp;
-    double      m_g;
-    double      m_ior;
+    Spectrum    m_reflectance;
+    Spectrum    m_mean_free_path;
+    double      m_mean_free_path_multiplier;
+    double      m_anisotropy;
+    double      m_from_ior;
+    double      m_to_ior;
+
+    // Precomputed values.
+    Spectrum    m_cdf;
+    double      m_max_mean_free_path;
+
+    // Convenience accessors for use after reflectance and
```

Comment:
Ah I understand what you mean: At some point `m_reflectance` and `m_mean_free_path` change meaning, and so it's clearer to use different identifiers.

I think it would be just cleaner (and no less efficient, really, since we don't store these values) to just introduce new members `m_sigma_a` and `m_sigma_s_prime`.